front/migrations/20240410_invert_agent_actions_configs_fkeys_step_1.ts

```diff
@@ -0,0 +1,116 @@
+import * as _ from "lodash";
+
+import {
+  AgentConfiguration,
+  AgentDustAppRunConfiguration,
+  AgentRetrievalConfiguration,
+  AgentTablesQueryConfiguration,
+} from "@app/lib/models";
+import logger from "@app/logger/logger";
+import { makeScript } from "@app/scripts/helpers";
+
+const backfillActionConfigs = async (execute: boolean) => {
+  const retrievalConfigs = await AgentRetrievalConfiguration.findAll({
+    where: {
+      agentConfigurationId: null,
+    },
+  });
+
+  logger.info(
+    `Found ${retrievalConfigs.length} retrieval configurations without agent configuration`
+  );
+
+  for (const chunk of _.chunk(retrievalConfigs, 16)) {
+    await Promise.all(
+      chunk.map(async (rc) => {
+        const agent = await AgentConfiguration.findOne({
+          where: {
+            retrievalConfigurationId: rc.id,
+          },
+        });
+        if (!agent) {
+          logger.warn(`No agent found for retrieval configuration ${rc.id}`);
```

Comment:
You mind find some in production. I know we have a few generationConfiguration in that situation

Reply:
You'll catch these once you try to enforce non nullity but we might want to decide what we want to do with them now?

Reply:
Yes there are some. Will do in the PR where I enforce non-nullable, IMO migration script is to delete those.